sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -914,6 +914,17 @@ static int sof_audio_probe(struct platform_device *pdev)
 
 	ssp_codec = sof_rt5682_quirk & SOF_RT5682_SSP_CODEC_MASK;
 
+	if (mach->mach_params.ssp_valid != false) {
```

Comment:
Why do you need an additional variable if you have a signed integer type? I suppose that negative values there are invalid by definition, no?

Reply:
remove this variable